pkg/controller/storageversionmigrator/util.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package storageversionmigrator
+
+import (
+	"fmt"
+	"strconv"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	corev1 "k8s.io/api/core/v1"
+	svmv1alpha1 "k8s.io/api/storagemigration/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func convertResourceVersionToInt(rv string) (int64, error) {
+	resourceVersion, err := strconv.ParseInt(rv, 10, 64)
+	if err != nil {
+		return 0, fmt.Errorf("failed to parse resource version %q: %w", rv, err)
+	}
+
+	return resourceVersion, nil
+}
+
+func getGVRFromResource(svm *svmv1alpha1.StorageVersionMigration) schema.GroupVersionResource {
+	return schema.GroupVersionResource{
+		Group:    svm.Spec.Resource.Group,
+		Version:  svm.Spec.Resource.Version,
+		Resource: svm.Spec.Resource.Resource,
+	}
+}
+
+// IsConditionTrue returns true if the StorageVersionMigration has the given condition
+// It is exported for use in tests
+func IsConditionTrue(svm *svmv1alpha1.StorageVersionMigration, conditionType svmv1alpha1.MigrationConditionType) bool {
+	return indexOfCondition(svm, conditionType) != -1
+}
+
+func indexOfCondition(svm *svmv1alpha1.StorageVersionMigration, conditionType svmv1alpha1.MigrationConditionType) int {
+	for i, c := range svm.Status.Conditions {
+		if c.Type == conditionType && c.Status == corev1.ConditionTrue {
+			return i
+		}
+	}
+	return -1
+}
+
+func setStatusConditions(
```

Comment:
While I think in the narrow case, this fits the immediate usage, prior to beta this needs a more "normal" method signature.  something like

```
setStatusCondition(&conditions, conditionType, status, reason, message){
// find existing to update in place
// no side effects on other conditinos
// if the status has not changed, don't change lastUpdateTime
// if the status has changed, set lastUpdateTime to now.
}

Reply:
Cool. I'll update setStatusCondition to above in beta